app/models/rental.rb

```diff
@@ -0,0 +1,18 @@
+class Rental < ApplicationRecord
+  before_validation :set_dates
```

Comment:
While I think this code works, given how `set_dates` is implemented, I do think there's probably a more direct way to achieve the same thing.

In this case, the logic in `set_dates` works out so that we're effectively setting _default_ values for the `checkout` and `due_date` columns. As such, we can use the [Rails 5 attributes API](https://karolgalanciak.com/blog/2016/12/04/introduction-to-activerecord-and-activemodel-attributes-api/):
```ruby
class Rental < ApplicationRecord
  attribute :checkout, :date, default: -> { Date.today }
  attribute :due_date, :date, default: -> { 7.days.from_now }

  ...

end
```